### PR TITLE
Report all solutions from a rule

### PR DESCRIPTION
**Pull request by damyan-ognyanov**
The plugin checks wheter a reported rule premice is the same of the targeget one and in such case ommit the solution. The rule may have more solutions so these must be reported. The PR fix is to enum all solutions from rule, exclude those that have a self referential premise and report those.

### CODE REVIEW

File: src/main/java/com/ontotext/trree/plugin/proof/ProofPlugin.java [hunk 1]
```diff
@@ -334,6 +336,42 @@ class Solution {
...
+					return false;
+				
+				if (other.premises.size() != this.premises.size())
+					return false;
+
+				//crosscheck
+				for (long[] p : this.premises) {
+					boolean exists = false;
+					for (long[] o : other.premises) {
+						if (o[0] == o[0] && o[1] == p[1] && o[2] == p[2] && o[3]== p[3] && o[4] == p[4]) {
```
Comment (avataar) [inline src/main/java/com/ontotext/trree/plugin/proof/ProofPlugin.java]: First condition should be o[0] == p[0].

File: src/test/java/com/ontotext/trree/plugin/proof/TestExplainWithProofPlugin.java [hunk 2]
```diff
@@ -88,5 +88,11 @@ public void testBasicInference() throws RepositoryException, MalformedQueryExcep
 			rep.shutDown();
 		}
 	}
+	
+	@Test
+	public void testFSIssueInference() throws RepositoryException, MalformedQueryException, QueryEvaluationException, RDFParseException, IOException {
```
Comment (avataar) [inline src/test/java/com/ontotext/trree/plugin/proof/TestExplainWithProofPlugin.java]: Empty test?

**Review by avataar — CHANGES_REQUESTED** [review]

File: src/main/java/com/ontotext/trree/plugin/proof/ProofPlugin.java (hunk 1 above)
Comment (damyan-ognyanov) [reply src/main/java/com/ontotext/trree/plugin/proof/ProofPlugin.java]: fixed

File: src/test/java/com/ontotext/trree/plugin/proof/TestExplainWithProofPlugin.java (hunk 2 above)
Comment (damyan-ognyanov) [reply src/test/java/com/ontotext/trree/plugin/proof/TestExplainWithProofPlugin.java]: removed the empty rest

**Review by avataar — APPROVED** [review]

**Pull request merged**